cypress/e2e/shared/fluxQueryBuilder.test.ts

```diff
@@ -491,9 +491,27 @@ describe('Script Builder', () => {
             'disabled'
           )
 
+          cy.log(
+            'does not diverge, when adding import statement outside of block'
+          )
+          cy.getByTestID('flux-toolbar-search--input')
+            .should('exist')
+            .type('fieldsAsCols')
+          cy.get('.flux-toolbar--list-item').first().click()
+          cy.getByTestID('flux-editor').contains('import')
+          cy.getByTestID('flux-sync--toggle').should(
+            'not.have.class',
+            'disabled'
+          )
+
+          /// FIXME: Does not work yet, since the LSP response for applyEdit starts at line 0
+          // cy.log(
+          //   'does not diverge, when further modifying block (with imports at top)'
+          // )
```

Comment:
Just curious, is this intended for another PR? If the test doesn't work yet, is there a bug?

Or it is done already? I tested on my local machine, it looks good

https://user-images.githubusercontent.com/14298407/196198408-09e08c87-1357-4853-b8a0-4b63f81a0fb7.mov

Reply:
It needs to be another PR. The remaining work (referenced in that `cy.log()`) is that if we add an import statement, then use the schemaBrowser again --> it will recreate the composition block at line 1.


https://user-images.githubusercontent.com/10232835/196234967-eac172d9-ab89-4f13-8a3a-b2c5eec7eb4d.mov




Update for that needs to be in the lsp, since it always returns an `applyEdit` message to the monaco-editor with the replacement line starting at the first line.

Reply:
Your question brings up an excellent point. This next bit of work (in the flux-lsp) should probably merge before this one does. Because the current prod (before this PR) just disables the sync once we use imports. So we probably want to leave that behavior until the lsp update goes in too.